datafusion/common/src/tree_node.rs

```diff
@@ -769,6 +770,263 @@ impl<T> Transformed<T> {
     }
 }
 
+/// [`Container`] contains elements that a function can be applied on or mapped. The
+/// elements of the container are siblings so the continuation rules are similar to
+/// [`TreeNodeRecursion::visit_sibling`] / [`Transformed::transform_sibling`].
+pub trait Container<'a, T: 'a>: Sized {
```

Comment:
I think it is a matter of preference, but what would you think about calling this something less generic, such as `TreeNodeContainer` ?

Reply:
Yeah, indeed that's a better name for container. Let me change it.

Reply:
Renamed in https://github.com/apache/datafusion/pull/13467/commits/e5aff72b1ea9af816b1465d3f8d09c0f42983a5b.